### ISSUE TITLE
f1 score is not improving during training

### ISSUE THREAD

**Issue by javak87** [seisbench/seisbench#156]
Dear Developers,
I tried to perform transfer learning using EQT, but I realize that the performance drops.
Inputs:
{
  "model": "EQTransformer",
  "base_path": "...",
  "visual_add_scal":"False",
  "visual_hist":false,
  "trainer_args": {
    "epochs": 20,
    "batch_size": 400,
    "num_workers": 4,
    "loss_weight":[0.05, 0.40, 0.55]
  },
  "model_args": {
    "lr": 1e-3,
    "base_model": "instance"
  },
  "augmentation":{
    "detection_fixed_window" : 2000,
    "sample_boundaries" : [null, null],
    "rotate_array" : false,
    "sigma" : 25
    }
  }

I used the this augmentation (https://github.com/seisbench/pick-benchmark/blob/95ec587a9ed04831a413e651cf04a5f271497738/benchmark/models.py#L376)

I also used this loss function (https://github.com/seisbench/pick-benchmark/blob/95ec587a9ed04831a413e651cf04a5f271497738/benchmark/models.py#L41)

To evaluate model, I utilized the EQT f1 core (https://github.com/smousavi05/EQTransformer/blob/2e99345516bbd155603fb7f08fc6266504fe5d83/EQTransformer/core/EqT_utils.py#L2082)

After training, I saw that the loss function drops, but the f1 score of det, P and S became worse than the initial state.
could you please tell me what is the problem?
![results](https://user-images.githubusercontent.com/44999772/209945836-400f8644-736c-4515-bd8d-7825c8a3bdc7.png)

**Comment by yetinam** [comment]
Hi @javak87 ,

I'm not sure if this is the issue, but I think you might be using the incorrect loss function. As there is no noise class, vector cross entropy will not converge against the correct limit. You'll need to use binary cross entropy separately on each output, i.e., P, S and detection. This is what [we do in the benchmark](https://github.com/seisbench/pick-benchmark/blob/95ec587a9ed04831a413e651cf04a5f271497738/benchmark/models.py#L404).

**Comment by javak87** [comment]
Dear @yetinam,
First of all thank you for your response.
I changed the loss function to binary cross entropy separately on each output, i.e., P, S and detection(exactly what you did).
I also used the **seisbench iquque dataset** for training.
The loss function continuedly decreasing, but f1 score of P and S is bouncing.
The loss weights for det, P, and S is [0.05, 0.40, 0.55]. As you can see the loss weight for detection is 0.05, resulting in small contribution to total loss. However, you can see  f1 score of P and S is bouncing,  but detection f1score is improving. 
Could you please tell me it is OK or some parts of training process go wrong?
Thanks

![binary_loss_results](https://user-images.githubusercontent.com/44999772/210170807-6ab421ed-5f87-4b67-903e-c56a6ba57dbe.png)

**Comment by yetinam** [comment]
Dear @javak87 ,
I'm afraid I can't tell you with certainty if your training is right now. The results look somewhat plausible, even though I'd expect better performance for P and S even on this dataset. Maybe the choice of threshold plays a role here too. In addition, I'd generally not recommend the Iquique dataset because the data quality is not too good. For a small but very high quality dataset I'd try ETHZ.

**Comment by javak87** [comment]
Dear @yetinam,
Sorry for the late response.
I used this augmentation that you used for EQT:
https://github.com/seisbench/pick-benchmark/blob/95ec587a9ed04831a413e651cf04a5f271497738/benchmark/models.py#L440
Could you please tell me which set of augmentations you used for EQT?
Thanks

**Comment by yetinam** [comment]
Hi,
sorry, this issue got lost. We used exactly the augmentations from the benchmark code you pointed to. Please note that the set of augmentations between training/validation/evaluation differs a bit.
